### ISSUE TITLE
VPC NAT Gateways limit is per-AZ

### ISSUE THREAD

**Issue by jantman** [jantman/awslimitchecker#214]
The VPC "NAT Gateways" limit merged in #204 and released in 0.5.1 is actually [per-AZ, not per-region](https://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_Appendix_Limits.html#vpc-limits-gateways). That code will need to be changed to add usage for each AZ separately.

This is considered a bug, as the limit isn't being evaluated properly.

The AWS docs call this limit "NAT gateways per Availability Zone" and say, "A NAT gateway in the pending, active, or deleting state counts against your limit." (default of 5, per AZ).

I'm torn on whether I should just fix the calculation issue and release this as 0.5.2, or fix the calculation and rename the limit from "NAT gateways" to "NAT gateways per Availability Zone", which would require holding it for 0.6.0.

**Comment by hltbra** [comment]
If you fix the bug the way I think you will (specifying a `resource_id` per AZ), the `get_current_limit()` will return the same value, but `get_current_usage()` will return a list of 1 or more elements, one per AZ, correct? Would you consider this as backward incompatible? If so, I'd say go with 0.6.0 anyway (and rename it). If not, I'd go with 0.5.2.

Sorry to create this problem for you.

**Comment by hltbra** [comment]
Similarly, I noticed that the metrics such as "Max spot instance requests per region" are implemented as a single value, instead of listing the usage per region.

```python
>>> c = AwsLimitChecker(region='us-east-1')
>>> c.find_usage("EC2", use_ta=False)
No handlers could be found for logger "awslimitchecker.services.ec2"
>>> ec2_limits = c.get_limits("EC2", use_ta=False)
>>> ec2_limits['EC2']['Max spot instance requests per region'].get_current_usage()
[<awslimitchecker.limit.AwsLimitUsage object at 0x10f5cb390>]
>>> ec2_limits['EC2']['Max spot instance requests per region'].get_current_usage()[0].resource_id is None
True
```

Would you consider this a bug?

**Comment by jantman** [comment]
@hltbra 
My gut reaction is to just call this 0.6.0 and change the name.

"Max spot instance requests per region" is implemented as a single value because awslimitchecker does not have the capability to check multiple regions; each AWSLimitChecker instance connects to a single region's endpoint, so there's no case where we can ever have data for multiple regions.

**Comment by jantman** [comment]
Fix merged to develop in #224

**Comment by jantman** [comment]
This has been released in 0.6.0 and is now live on PyPI
